src/main/java/jbash/commands/CmdLs.java

```diff
@@ -19,31 +18,31 @@ public String getHelp() {
     }
 
     @Override
-    public int execute(List<String> args) {
+    public int execute(List<String> argv) {
         FileSystemAPI FSAPI = FileSystemAPI.getInstance();
         boolean multiplePaths = false;
-        if (args.isEmpty()) {
+        if (argv.isEmpty()) {
             List<FileSystemObject> currentFiles = FSAPI.getCurrentDirectory().getChildren();
             if (currentFiles.isEmpty()) { return 0; }
             for (FileSystemObject file : currentFiles) {
-                System.out.print(file.getName());
-                System.out.print(" ");
+                cmdPrint(file.getName());
+                cmdPrint(" ");
             }
-            System.out.println();
+            cmdPrintln();
         } else {
-            if (args.size() > 1) { multiplePaths = true; }
-            for (String path : args) {
+            if (argv.size() > 1) { multiplePaths = true; }
+            for (String path : argv) {
                 Optional<FileSystemObject> directory = FSAPI.getFileSystemObject(path);
-                if (directory.isEmpty()) { System.out.println("ls: cannot access '" + path + "': No such file or directory"); }
-                else if (!(directory.get() instanceof Directory)) { System.out.println(directory.get().getName()); }
+                if (directory.isEmpty()) { cmdErrln("ls: cannot access '" + path + "': No such file or directory"); }
+                else if (!(directory.get() instanceof Directory)) { cmdPrintln(directory.get().getName()); }
```

Comment:
as we discussed before, eventually we can probably remove all the `instanceof` nonsense and use the new getDirectory() method

Reply:
Right. I much prefer the syntax of `getDirectory().orElse(null)` and then checking for `null`.

Reply:
its much cleaner